### PR TITLE
applications: Fix configuration loading

### PR DESCRIPTION
**Pull request by nordicjm**
Fixes configuration loading, needed for overlaying application configuration to function correctly.

### CODE REVIEW

**Review by kkasperczyk-no — APPROVED** [review]

File: applications/nrf_desktop/CMakeLists.txt [hunk 1]
```diff
@@ -6,26 +6,24 @@
...
-
+set(APPLICATION_CONFIG_DIR "${CMAKE_CURRENT_SOURCE_DIR}/configuration/\${BOARD}")
 
 find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
 project("nRF52 Desktop"
         VERSION 0.1)
 
 ################################################################################
 
+if (NOT EXISTS "${APPLICATION_CONFIG_DIR}")
```
Comment (MarekPieta) [inline applications/nrf_desktop/CMakeLists.txt:20]: We already pass the `APPLICATION_CONFIG_DIR` to Zephyr build system before the actual check. Can the fatal error be reported there in this case?

Comment (MarekPieta) [reply applications/nrf_desktop/CMakeLists.txt:20]: Also from what I understood we expect that Zephyr build system may modify content of the `APPLICATION_CONFIG_DIR` (that's why the check was moved). Consider logging the actually used configuration directory (`APPLICATION_CONFIG_DIR`) in that case.

Comment (nordicjm) [reply applications/nrf_desktop/CMakeLists.txt:20]: It already does, e.g.: `Merged configuration '/tmp/bb/nrf/applications/nrf_desktop/configuration/nrf52840dk_nrf52840/prj.conf'`

File: applications/zigbee_weather_station/CMakeLists.txt [hunk 2]
```diff
@@ -6,18 +6,10 @@
 
 cmake_minimum_required(VERSION 3.20.0)
 
-
-if (NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/configuration/${BOARD}")
```
Comment (tomchy) [inline applications/zigbee_weather_station/CMakeLists.txt:10]: Could you explain, why this check is considered as an issue?

**Review by tomchy — CHANGES_REQUESTED** [review]

File: applications/zigbee_weather_station/CMakeLists.txt (hunk 2 above)
Comment (nordicjm) [reply applications/zigbee_weather_station/CMakeLists.txt:10]: Zephyr already checks that APPLICATION_CONFIG_DIR exists and will error out if it does not

**Review by tomchy — APPROVED** [review]

**Pull request merged**